src/go/configgenerator/listener_generator_test.go

```diff
@@ -71,9 +71,86 @@ func TestTranscoderFilter(t *testing.T) {
       "autoMapping":true,
       "convertGrpcStatus":true,
       "ignoredQueryParameters":[
+         "access_token",
          "api_key",
+         "key"
+      ],
+      "protoDescriptorBin":"%s",
+      "services":[
+         "%s"
+      ]
+   }
+}
+      `, fakeProtoDescriptor, testApiName),
+		},
+		{
+			desc: "Success. Generate transcoder filter with custom apiKey locations and custom jwt locations",
+			fakeServiceConfig: &confpb.Service{
+				Name: testProjectName,
+				Apis: []*apipb.Api{
+					{
+						Name: testApiName,
+					},
+				},
+				SourceInfo: &confpb.SourceInfo{
+					SourceFiles: []*anypb.Any{content},
+				},
+				Authentication: &confpb.Authentication{
+					Providers: []*confpb.AuthProvider{
+						{
+							Id:      "auth_provider",
+							Issuer:  "issuer-0",
+							JwksUri: "https://fake-jwks.com",
+							JwtLocations: []*confpb.JwtLocation{
+								{
+									In: &confpb.JwtLocation_Header{
+										Header: "jwt_query_header",
+									},
+									ValuePrefix: "jwt_query_header_prefix",
+								},
+								{
+									In: &confpb.JwtLocation_Query{
+										Query: "jwt_query_param",
+									},
+								},
+							},
+						},
+					},
+				},
+				SystemParameters: &confpb.SystemParameters{
+					Rules: []*confpb.SystemParameterRule{
+						{
+							Selector: testApiName,
+							Parameters: []*confpb.SystemParameter{
+								{
+									Name:              "api_key",
+									HttpHeader:        "header_name_1",
+									UrlQueryParameter: "query_name_1",
+								},
+								{
+									Name:              "api_key",
+									HttpHeader:        "header_name_2",
+									UrlQueryParameter: "query_name_2",
+								},
+							},
+						},
+					},
+				},
+			},
+			wantTranscoderFilter: fmt.Sprintf(`
+{
+   "name":"envoy.grpc_json_transcoder",
+   "typedConfig":{
+      "@type":"type.googleapis.com/envoy.config.filter.http.transcoder.v2.GrpcJsonTranscoder",
+      "autoMapping":true,
+      "convertGrpcStatus":true,
+      "ignoredQueryParameters":[
+         "access_token",
+         "api_key",
+         "jwt_query_param",
          "key",
```

Comment:
Why is `access_token`, `api_key`, and `key` still here? It's not in the system params or jwt locations? I thought we would not use the default values if user overrides jwt locations.